ee/clickhouse/views/insights.py

```diff
@@ -125,10 +126,22 @@ def funnel_correlation(self, request: Request, *args: Any, **kwargs: Any) -> Res
 
     @cached_function
     def calculate_funnel_correlation(self, request: Request) -> Dict[str, Any]:
+        # Put all the request into the existing filter class, so we isolate out
+        # changes to just clarify what is relevant to correlation
+        funnel_filter = Filter(request=request, team=self.team)
+        # Package the funnel_filter up, along with the specific funnel
+        # correlation request params
+        # TODO: further dispatch property and event requests separately
+        # TODO: Define CorrelationPerson request/dispatcher
+        correlation_request = FunnelCorrelationRequest(
```

Comment:
The annoying bit here is how this gets separated from `get_filter` , which is used in all caching.

See this for example: https://github.com/PostHog/posthog/issues/6658

I think this making-interface-clearer is a great idea! But we ought to:

1. Either use `get_filter` everywhere, so there's no discrepancies between caching filters and actual request filters
2. Do all filter splitting inside respective classes (so, inside `FunnelCorrelation` here).

Reply:
Yes great point. Like you say, this isn't touching anything but the `FunnelCorrelation` interface. Do you see any issue with that specifically?

On point 1. the simplest thing I can see is that we, rather than using `get_filter` everywhere, just perform the passing once and pass the resulting filter/request through. Would this handle the referenced error?

As for point 2. yeah I think `FunnelCorrelation` probably has a better idea of what can be cached for sure.

I'll leave those as separate task if thats ok?

Reply:
Nope, very much do like the idea of cleaning up the interface. Makes it clearer for whoever comes next